x-pack/plugin/mapper-constant-keyword/src/test/java/org/elasticsearch/xpack/constantkeyword/mapper/ConstantKeywordFieldMapperTests.java

```diff
@@ -333,6 +333,17 @@ public void testNullValueSyntheticSource() throws IOException {
         assertThat(syntheticSource(mapper, b -> {}), equalTo("{}"));
     }
 
+    public void testNoValueInDocumentSyntheticSource() throws IOException {
```

Comment:
Maybe in a follow up we can extend `DataGenerationHelper` to use constant keyword? That we would test this with reindex too in `ReindexChallengeRestIT` subclasses.